cardano-testnet/src/Testnet/Start/Types.hs

```diff
@@ -34,37 +36,61 @@ import           Testnet.Filepath
 import           Hedgehog (MonadTest)
 import qualified Hedgehog.Extras as H
 
+-- | Command line options for the @cardano-testnet@ executable. They are used
+-- in the parser, and then get split into 'CardanoTestnetOptions' and
+-- 'ShelleyTestnetOptions'
+data CardanoTestnetCliOptions = CardanoTestnetCliOptions
+  { cliTestnetOptions :: CardanoTestnetOptions
```

Comment:
What is the difference between these two types? We should not have two types that describe the testnet options.

Reply:
@Jimbo4350> the difference is explained in the haddocks of the definitions of those types:

```hs
-- | Options which, contrary to 'ShelleyTestnetOptions' are not implemented
-- by tuning the genesis files.
data CardanoTestnetOptions = CardanoTestnetOptions
```

and

```hs
-- | Options that are implemented by writing fields in the Shelley genesis file.
data ShelleyTestnetOptions = ShelleyTestnetOptions
```